alsoHeresTheInterviewQuestionIHad/alsoHeresTheInterviewQuestionIHadMySolution.js

```diff
@@ -0,0 +1,96 @@
+Illustration of first example
+
+
+Given H = [5, 3, 2, 4], the function should return 17. The result can be achieved by covering the first building with a banner of size 5×1 and the other buildings with a banner of size 4×3:
+Illustration of second example
+Given H = [5, 3, 5, 2, 1], your function should return 19. The result can be achieved by covering the first three buildings with a banner of size 5×3 and the other two with a banner of size 2×2:
+Illustration of third example
+
+
+Given H = [7, 7, 3, 7, 7], your function should return 35. The result can be achieved by using one banner of size 7×5:
+Illustration of fourth example
+
+
+Given H = [1, 1, 7, 6, 6, 6], your function should return 30. The result can be achieved by using banners of size 1×2 and 7×4:
+Illustration of fifth example
+
+
+Write an efficient algorithm for the following assumptions:
+
+N is an integer within the range [1..100,000];
+each element of array H is an integer within the range [1..10,000]
+
+*/
+
+const fs = require("fs");
+const buildings = fs.readFileSync("data.txt", "utf-8").split(",");
+
+function bestArea(streetArray) {
+    let numberOfBuildings = streetArray.length;
+    const buildingAreasLeft = [];
+    const buildingAreasRight = [];
+
+    let tallestBuildingFromLeft = 0;
+    let tallestBuildingFromRight = 0;
+
+    let bannerOneAreaRight = 0;
+    let bannerTwoAreaLeft = 0;
+    let bannerOneAreaLeft = 0;
+    let bannerTwoAreaRight = 0;
+    let bestAreaInFunction = 0;
+
+    for (let i = 0; i < numberOfBuildings; i++) {
+        if (streetArray[i] > tallestBuildingFromLeft) {
+            tallestBuildingFromLeft = streetArray[i];
+        }
+
+        if (streetArray[numberOfBuildings - i - 1] > tallestBuildingFromRight) {
+            tallestBuildingFromRight = streetArray[numberOfBuildings - i - 1];
+        }
+
+        buildingAreasLeft[i] = tallestBuildingFromLeft * (i + 1);
+        buildingAreasRight[numberOfBuildings - i - 1] = tallestBuildingFromRight * (i + 1);
+
+        if (i > (numberOfBuildings - 1) / 2) {
+            bannerOneAreaLeft = buildingAreasLeft[numberOfBuildings - i - 1];
+            bannerTwoAreaLeft = buildingAreasRight[numberOfBuildings - i];
+            bannerOneAreaRight = buildingAreasLeft[i - (numberOfBuildings % 2)];
+            bannerTwoAreaRight = buildingAreasRight[i + 1 - (numberOfBuildings % 2)];
+
+            if (bestAreaInFunction === 0 || bestAreaInFunction > bannerOneAreaLeft + bannerTwoAreaLeft) {
+                bestAreaInFunction = bannerOneAreaLeft + bannerTwoAreaLeft;
+            }
+
+            if (bestAreaInFunction === 0 || bestAreaInFunction > bannerOneAreaRight + bannerTwoAreaRight) {
+                bestAreaInFunction = bannerOneAreaRight + bannerTwoAreaRight;
+            }
+        }
+    }
+    return bestAreaInFunction;
+}
+
+bestArea(buildings);

```

Comment:
don't need this

Reply:
You already approved of this code?  It's my original solution, and I thought it wasn't even up for further discussion until Eric had picked at it?

Reply:
I'm not perfect :( but I am working on it